collectors/python.d.plugin/edgex/edgex.chart.py

```diff
@@ -0,0 +1,312 @@
+try:
+    from Queue import Queue
+except ImportError:
+    from queue import Queue
+
+from bases.FrameworkServices.UrlService import UrlService
+
+METHODS = namedtuple('METHODS', ['get_data', 'url', 'run'])
+
+ORDER = [
+    'events_throughput',
+    'readings_throughput',
+    # 'events_count',
+    'devices_number',
+    'memory_alloc',
+    'memory_malloc',
+    'memory_frees',
+    'memory_liveObjects'
+]
+
+# default module values (can be overridden per job in `config`)
+update_every = 5
+
+CHARTS = {
+    'events_throughput':{
+        'options': [None, 'Events and Readings (readings are a certain type of event) per second in the EdgeX platform', 'events/s',
+                    'events throughput','edgex.events', 'line'],
+        'lines': [
+            ['events', None, 'incremental'],
+            ['readings', None, 'incremental']
+        ]
+    },
+    # 'readings_throughput':{
+    #     'options': [None, 'Readings per second in the EdgeX platform', 'readings/s', 'throughput', 'edgex.readings',
+    #                 'line'],
+    #     'lines': [
+    #         ['readings', None, 'incremental']
+    #     ]
+    # },
+    'events_count':{
+        'options': [None, 'Total number of events and readings in the EdgeX platform', 'events', 'events count', 'edgex.events_readings_abs',
+                    'line'],
+        'lines': [
+            ['readings', None, 'absolute'],
+            ['events', None, 'absolute']
+        ]
+    },
+    'devices_number':{
+        'options': [None, 'Number of registered devices in the EdgeX platform', 'devices', 'devices count', 'edgex.devices_number',
+                    'line'],
+        'lines': [
+            ['registered_devices', None, 'absolute'],
+        ]
+    },
+    'memory_alloc':{
+        'options': [None, 'Alloc: currently allocated number of bytes on the heap of each EdgeX service. Provided by the Golang runtime Package.',
+                    'bytes', 'memory metrics', 'edgex.memory_alloc', 'line'],
+        'lines': [
+            ['core_data_alloc', None, 'absolute'],
+            ['core_metadata_alloc', None, 'absolute'],
+            ['core_command_alloc', None, 'absolute'],
+            ['support_logging_alloc', None, 'absolute']
+        ]
+    },
+    'memory_malloc':{
+        'options': [None, 'Mallocs: it is the cumulative count of heap objects allocated. Provided by the Golang runtime Package.', 'heap objects',
+                    'memory metrics','edgex.memory_malloc', 'line'],
+        'lines': [
+            ['core_data_malloc', None, 'absolute'],
+            ['core_metadata_malloc', None, 'absolute'],
+            ['core_command_malloc', None, 'absolute'],
+            ['support_logging_malloc', None, 'absolute']
+        ]
+    },
+    'memory_frees':{
+        'options': [None, 'Frees: it is the cumulative count of heap objects freed. Provided by the Golang runtime Package.', 'heap objects',
+                    'memory metrics', 'edgex.memory_frees', 'line'],
```

Comment:
~~Provided by the Golang runtime Package.~~

> it is the cumulative count of heap objects freed.

If these counters never reset these chart doesnt look meaningful. Probably we need rate? (absolute->incremental)

Reply:
To be absolutely honest, since I am not familiar with Go and the underline usage of computer resources, I wasn't too certain what info was meaningful and what not.

can you take a look here? https://golang.org/pkg/runtime/#MemStats